src/game/paintings.c

```diff
@@ -1088,188 +952,170 @@ void reset_painting(struct Painting *painting) {
-void move_ddd_painting(struct Painting *painting, UNUSED f32 frontPos, f32 backPos, UNUSED f32 speed) {
-    painting->posX = backPos;
-    gDddPaintingStatus = (DDD_FLAG_BOWSERS_SUB_BEATEN | DDD_FLAG_BACK);
-}
-#else
-void move_ddd_painting(struct Painting *painting, f32 frontPos, f32 backPos, f32 speed) {
-    // Obtain the DDD star flags
-    u32 dddFlags = save_file_get_star_flags(gCurrSaveFileNum - 1, COURSE_NUM_TO_INDEX(COURSE_DDD));
-    // Get the other save file flags
-    u32 saveFileFlags = save_file_get_flags();
-    // Find out whether Board Bowser's Sub was collected
-    u32 bowsersSubBeaten = dddFlags & BOARD_BOWSERS_SUB;
-    // Check whether DDD has already moved back
-    u32 dddBack = saveFileFlags & SAVE_FLAG_DDD_MOVED_BACK;
-
-    if (!bowsersSubBeaten && !dddBack) {
-        // If we haven't collected the star or moved the painting, put the painting at the front
-        painting->posX = frontPos;
-        gDddPaintingStatus = DDD_FLAGS_NONE;
-    } else if (bowsersSubBeaten && !dddBack) {
-        // If we've collected the star but not moved the painting back,
-        // Each frame, move the painting by a certain speed towards the back area.
-        painting->posX += speed;
-        gDddPaintingStatus = DDD_FLAG_BOWSERS_SUB_BEATEN;
-        if (painting->posX >= backPos) {
-            painting->posX = backPos;
-            // Tell the save file that we've moved DDD back.
-            save_file_set_flags(SAVE_FLAG_DDD_MOVED_BACK);
-        }
-    } else if (bowsersSubBeaten && dddBack) {
-        // If the painting has already moved back, place it in the back position.
-        painting->posX = backPos;
-        gDddPaintingStatus = (DDD_FLAG_BOWSERS_SUB_BEATEN | DDD_FLAG_BACK);
-    }
-}
+    obj->oPosX = backPos;
+    return;
 #endif
-
-/**
- * Set the painting's node's layer based on its alpha
- */
-void set_painting_layer(struct GraphNodeGenerated *gen, struct Painting *painting) {
-    if (painting->alpha == 0xFF) { // Opaque
-        SET_GRAPH_NODE_LAYER(gen->fnNode.node.flags, LAYER_OCCLUDE_SILHOUETTE_OPAQUE);
+    // Obtain the DDD star flags and find out whether Board Bowser's Sub was collected.
+    if (save_file_get_star_flags((gCurrSaveFileNum - 1), COURSE_NUM_TO_INDEX(COURSE_DDD)) & STAR_FLAG_ACT_1) {
+        // Get the other save file flags and check whether DDD has already moved back.
+        if (save_file_get_flags() & SAVE_FLAG_DDD_MOVED_BACK) {
+            // If the painting has already moved back, place it in the back position.
+            obj->oPosX = backPos;
+        } else {
+            // If we've collected the star but not moved the painting back,
+            // Each frame, move the painting by a certain speed towards the back area.
+            obj->oPosX += speed;
+            if (obj->oPosX >= backPos) {
+                obj->oPosX = backPos;
+                // Tell the save file that we've moved DDD back.
+                save_file_set_flags(SAVE_FLAG_DDD_MOVED_BACK);
```

Comment:
I just sorta realized, I don't honestly think there's any reason for this to be a save flag whatsoever. The animation for the painting moving back does not need to rely on checking the save file; you can just set a global when loading the save file that checks whether the star has been collected and set the initial painting position that way. I don't see a reason why the flag wouldn't be set but the star is collected in any case other than immediately after ejecting from the painting (or with nonstop stars enabled or something but that would also behave weirdly with vanilla behavior anyway).

Reply:
Done. I ended up setting the global when the painting is forward and the star isn't collected instead of checking the save file on game load.